model/config.go

```diff
@@ -2314,14 +2318,15 @@ func (o *Config) SetDefaults() {
 		}
 	}
 
+	isNew := o.isNew()
```

Comment:
Let's move this to the top of the method (assigning to `o.isNew` as discussed above) in case ldap settings or saml settings need to make decisions about this in the future.